### PR TITLE
Add ability to send events with absolute values to graphite

### PR DESCRIPTION
**Pull request by azat**
Category (leave one):
- Improvement

Short description (up to few sentences):

Add ability to send profile events (counters) with cumulative values to graphite. It can be enabled under `<events_cumulative>` in server `config.xml`.

Detailed description (optional):

Before this patch only delta values for events was sent, while this is not convenient for calculating rates. So after this patch <events_cumulative> is introduced.

### CODE REVIEW

**Comment by azat** [comment]
> Style check — Found 4 errors

Should be fixed in 1f34c44521

**Comment by azat** [comment]
>Style check — Found 1 errors

Fixed

File: dbms/programs/server/MetricsTransmitter.h [hunk 1]
```diff
@@ -43,14 +44,15 @@ class MetricsTransmitter
...
+    bool send_events_deltas;
     bool send_events;
     bool send_metrics;
     bool send_asynchronous_metrics;
 
     bool quit = false;
     std::mutex mutex;
     std::condition_variable cond;
-    ThreadFromGlobalPool thread{&MetricsTransmitter::run, this};
+    std::unique_ptr<ThreadFromGlobalPool> thread;
```
Comment (alexey-milovidov) [inline dbms/programs/server/MetricsTransmitter.h]: std::optional can suite better.

File: dbms/programs/server/MetricsTransmitter.cpp [hunk 2]
```diff
@@ -5,29 +5,58 @@
...
     send_metrics = config.getBool(config_name + ".metrics", true);
     send_asynchronous_metrics = config.getBool(config_name + ".asynchronous_metrics", true);
+
+    const std::string events_key = config_name + ".events";
+    try
+    {
+        send_events = config.getBool(events_key, true);
+        send_events_deltas = true;
+    }
+    catch (const Poco::SyntaxException &)
```
Comment (alexey-milovidov) [inline dbms/programs/server/MetricsTransmitter.cpp]: Still don't like it.

**Comment by alexey-milovidov** [comment]
Let's add `<events_absolute>` that can be enabled and disabled independent with `<events>`.
You can even enable both.

Absolute values should go to `ProfileEventsAbsolute` (or `ProfileEventsAbs`) path.

**Comment by azat** [comment]
>Let's add <events_absolute> that can be enabled and disabled independent with <events>.

Done

**Comment by alexey-milovidov** [comment]
Ok forgot that the word "cumulative" is better than "absolute" :(
Will change after merge.

**Review by alexey-milovidov — APPROVED** [review]
Great! Now it's more simple and better.

**Comment by alexey-milovidov** [comment]
70bc895

**Comment by azat** [comment]
>70bc895

great, thanks!

**Pull request merged**